### PR TITLE
metrics: add offline region metrics

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Related to #2997.

### What is changed and how it works?

This PR adds metrics of regions' health for those offline stores and move the offline info from the stats to offlineStats

### Check List

Tests

- Unit test

### CODE REVIEW

File: server/statistics/region_collection.go [hunk 1]
```diff
@@ -100,43 +129,47 @@ func (r *RegionStatistics) Observe(region *core.RegionInfo, stores []*core.Store
...
 	}
 
-	if len(region.GetPeers()) < desiredReplicas {
-		r.stats[MissPeer][regionID] = region
-		peerTypeIndex |= MissPeer
-	} else if len(region.GetPeers()) > desiredReplicas {
-		r.stats[ExtraPeer][regionID] = region
-		peerTypeIndex |= ExtraPeer
-	}
+	var isOffline bool
```
Comment (HunDunDM) [inline server/statistics/region_collection.go:132]:
```suggestion
	var hasOffline bool
```

Comment (rleungx) [reply server/statistics/region_collection.go:132]: IMO, isOffline is more suitable here?

File: server/statistics/region_collection.go [hunk 2]
```diff
@@ -100,43 +129,47 @@ func (r *RegionStatistics) Observe(region *core.RegionInfo, stores []*core.Store
...
 
-	if len(region.GetDownPeers()) > 0 {
-		r.stats[DownPeer][regionID] = region
-		peerTypeIndex |= DownPeer
+	for _, store := range stores {
+		if store.IsOffline() {
+			peer := region.GetStorePeer(store.GetID())
+			if peer != nil {
+				isOffline = true
```
Comment (lhy1024) [inline server/statistics/region_collection.go:138]:
```suggestion
				isOffline = true
				break
```

**Review by lhy1024 — APPROVED** [review]

File: server/statistics/region_collection.go (hunk 1 above)
Comment (HunDunDM) [reply server/statistics/region_collection.go:132]: The region is offline, which looks like the region is not available. But I think maybe we should replace `offline`?

**Comment by AndreMouche** [comment]
LGTM

**Review by HunDunDM — APPROVED** [review]

**Pull request merged**